GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/WorkingDirectoryTests.cs

```diff
@@ -541,11 +541,25 @@ public void TruncatedObjectRedownloaded()
             objectPath.ShouldBeAFile(this.fileSystem);
             new FileInfo(objectPath).Length.ShouldEqual(objectLength - 16);
 
-            // Read the original path and verify its contents are correct
-            this.Enlistment.GetVirtualPathTo("Test_EPF_WorkingDirectoryTests", "TruncatedObjectRedownloaded.txt").ShouldBeAFile(this.fileSystem).WithContents(testFileContents);
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                // Mac can't correct corrupt objects, but it should detect them and add to ModifiedPaths.dat

```

Comment:
Should we also validate that the file is reported as modified by git status?

Reply:
Looks like the best way to do this is CheckGitCommandAgainstGVFSRepo (Since we're not in GitCommands).  Let me know if that is incorrect.